### PR TITLE
RT-507:Loading indicators

### PR DESCRIPTION
**Pull request by yashpmohod**
## Overview

  Added a simple progress bar for layer loading, it shows until all layer is loaded and then after loading all layer it will hide  

## How it was tested
  Tested on local

## Checklist

- [x] The title contains a short meaningful summary
- [x] I have added a link to this PR in the Jira issue
- [x] I have described how this was tested
- [x] I have included screen shots for changes that affect the user interface
- [ ] I have updated unit tests
- [x] I have run unit tests locally
- [ ] I have updated documentation (including README)

### CODE REVIEW

**Comment by yashpmohod** [comment]
Peek 2022-12-21 12-23

**Review by amolht — APPROVED** [review]

File: projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.module.ts [hunk 1]
```diff
@@ -7,10 +7,11 @@ import { MatInputModule } from '@angular/material/input';
...
 import { MatSortModule } from '@angular/material/sort';
 import { MatButtonModule } from '@angular/material/button';
-
+import { MatProgressBarModule } from '@angular/material/progress-bar';
 @NgModule({
   declarations: [DataLayerBrowserComponent],
-  imports: [CommonModule, MatIconModule, MatTableModule, MatSortModule, MatInputModule, MatButtonModule, ReactiveFormsModule],
+  imports: [CommonModule, MatIconModule, MatTableModule, MatSortModule, MatInputModule, MatButtonModule, ReactiveFormsModule, MatProgressBarModule],
   exports: [DataLayerBrowserComponent],
+  schemas: [CUSTOM_ELEMENTS_SCHEMA]
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.module.ts:15]: why is this needed?

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts [hunk 2]
```diff
@@ -55,7 +55,10 @@ export class DataLayerManagerService {
   private timelineRangeSubject = new ReplaySubject<{ min: number; max: number }>();
   timelineRange$ = this.timelineRangeSubject.pipe(throttleTime(100, undefined, { leading: true, trailing: true }));
 
+  loader$ = new BehaviorSubject<boolean>(false); 
   loading = false;
+  loadingObservable = new BehaviorSubject<boolean>(false);
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts:60]: why do we have two subjects for the same thing? I prefer "loading" over "loader"

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts [hunk 3]
```diff
@@ -69,16 +72,17 @@ export class DataLayerManagerService {
...
+    this.loader$.next(true);
     merge(...this.dataLayerServices.map((service) => service.retrieve())).subscribe({
       next: (layer) =>
         this.stateSubject.next({
           ...this.stateSubject.value,
           layers: this.mergeService.merge(this.state.layers, layer),
         }),
       error: (err) => console.error(err),
-      complete: () => (this.loading = false),
+      complete: () => {this.loading = false; this.loader$.next(false);},
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts:83]: you can probably remove the boolean `loading` property. I don't think it's used anywhere.

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import { forwardRef, Inject, Injectable, NgZone } from '@angular/core';
-import { BehaviorSubject, distinctUntilChanged, map, merge, Observable, ReplaySubject, throttleTime } from 'rxjs';
+import { BehaviorSubject, distinctUntilChanged, map, merge, Observable, of, ReplaySubject, throttleTime } from 'rxjs';
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts:2]: unused import?

File: projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.component.ts [hunk 5]
```diff
@@ -32,6 +34,10 @@ export class DataLayerBrowserComponent implements OnInit, AfterViewInit {
       layer.category?.toLowerCase().includes(filter) ||
       layer.datasets.some((dataset) => dataset.label?.toLowerCase().includes(filter));
     this.filterControl.valueChanges.subscribe((value) => (this.dataSource.filter = value?.trim().toLowerCase() ?? ''));
+
+    this.layerManager.loader$.subscribe((loading:boolean) => {
+       this.isProgress = loading;
+    })
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.component.ts:40]: this seems unnecessary. can we just use async pipe?

File: projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.component.html [hunk 6]
```diff
@@ -4,11 +4,23 @@ <h2>Available Data Layers</h2>
...
-      <button mat-icon-button matSuffix (click)="filterControl.setValue('')"><mat-icon fontIcon="close"></mat-icon></button>
+      <button mat-icon-button matSuffix (click)="filterControl.setValue('')"><mat-icon
+          fontIcon="close"></mat-icon></button>
     </mat-form-field>
   </header>
+  <section class="progress-bar-section">
+    <mat-progress-bar *ngIf="isProgress"
+        class="progress-bar-margin"
+        mode="indeterminate"
+        [value]="value">
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.component.html:15]: why are we passing a value to indeterminate progress bar?

**Review by semicolin — CHANGES_REQUESTED** [review]

File: projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.module.ts (hunk 1 above)
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.module.ts]: I have remove that schemas.

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts (hunk 2 above)
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts]: Yes ,I am forgot to remove that lodingObservable that was previously use.

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts (hunk 3 above)
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts]: Yes I was remove that boolean loading property i thought it we are using that.

File: projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts (hunk 4 above)
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/data-layer/data-layer-manager.service.ts]: Yes ,I am remove all unused import from that file.

File: projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.component.ts (hunk 5 above)
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.component.ts]: Yes, I am use async pipe instead of this.

File: projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.component.html (hunk 6 above)
Comment (yashpmohod) [reply projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.component.html]: I am remove the value.

File: projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.component.spec.ts [hunk 7]
```diff
@@ -15,6 +15,7 @@ const mockLayerManager = {
   availableLayers$: EMPTY,
   selectedLayers$: EMPTY,
   timelineRange$: EMPTY,
+  loader$: EMPTY
```
Comment (semicolin) [inline projects/ngx-charts-on-fhir/src/lib/data-layer-browser/data-layer-browser.component.spec.ts:18]: this doesn't match the real property name

**Review by semicolin — CHANGES_REQUESTED** [review]

**Review by semicolin — APPROVED** [review]
Looks good!

**Review by amolht — APPROVED** [review]
